diff --git a/pkgs/development/compilers/gcc/common/libgcc.nix b/pkgs/development/compilers/gcc/common/libgcc.nix
index a7de840adc8d..42edea641ec0 100644
--- a/pkgs/development/compilers/gcc/common/libgcc.nix
+++ b/pkgs/development/compilers/gcc/common/libgcc.nix
@@ -7,6 +7,7 @@
 , enableShared
 , targetPlatform
 , hostPlatform
+, buildPackages
 , withoutTargetLibc
 , libcCross
 }:
@@ -156,5 +157,8 @@ in
     #
     + ''
       patchelf --set-rpath "" $libgcc/lib/libgcc_s.so.${libgcc_s-version-major}
+    ''
+    + lib.optionalString (targetPlatform != hostPlatform) ''
+      ${lib.getExe' buildPackages.binutils-unwrapped-all-targets "strip"} --strip-debug $libgcc/lib/libgcc_s.so.${libgcc_s-version-major}
     '');
 }))]))
